core/src/impl/Kokkos_hwloc.cpp

```diff
@@ -162,7 +162,7 @@ unsigned thread_mapping(const char* const label, const bool allow_async,
   const unsigned threads_per_core =
       thread_count / (use_numa_count * use_cores_per_numa);
 
-  if (thread_spawn_synchronous) {
+  if (thread_spawn_synchronous != 0) {
```

Comment:
Why not convert `thread_spawn_synchronous` to a `bool`? It's assigned 0 or 1 just above based on a ternary and some conditions.